dev_tools/utils/nxdl_utils.py

```diff
@@ -260,59 +259,35 @@ def belongs_to_capital(params):
 
 def get_local_name_from_xml(element):
     """Helper function to extract the element tag without the namespace."""
-    return element.tag[element.tag.rindex("}") + 1 :]
+    return remove_namespace_from_tag(element.tag)
 
 
 def get_own_nxdl_child_reserved_elements(child, name, nxdl_elem):
     """checking reserved elements, like doc, enumeration"""
-    if get_local_name_from_xml(child) == "doc" and name == "doc":
-        if nxdl_elem.get("nxdlbase"):
-            child.set("nxdlbase", nxdl_elem.get("nxdlbase"))
-            child.set("nxdlbase_class", nxdl_elem.get("nxdlbase_class"))
-            child.set("nxdlpath", nxdl_elem.get("nxdlpath") + "/doc")
-        return child
-    if get_local_name_from_xml(child) == "enumeration" and name == "enumeration":
-        if nxdl_elem.get("nxdlbase"):
-            child.set("nxdlbase", nxdl_elem.get("nxdlbase"))
-            child.set("nxdlbase_class", nxdl_elem.get("nxdlbase_class"))
-            child.set("nxdlpath", nxdl_elem.get("nxdlpath") + "/enumeration")
-        return child
+    local_name = get_local_name_from_xml(child)
+    if local_name == "doc" and name == "doc":
+        return set_nxdlpath(child, nxdl_elem, tag_name=name)
+
+    if local_name == "enumeration" and name == "enumeration":
+        return set_nxdlpath(child, nxdl_elem, tag_name=name)
     return False
```

Comment:
inconsistent return value types, yeah possible in python but poor design in my opinion, either you get a path (string) or a boolean, but not a boolean or a string, better return None if no path was resolvable